sdk/src/asset_handlers/sfnt_io.rs

```diff
@@ -1302,13 +1259,13 @@ fn read_c2pa_from_stream<T: Read + Seek + ?Sized>(reader: &mut T) -> Result<Tabl
     let sfnt = SfntFont::from_reader(reader).map_err(|_| Error::FontLoadError)?;
     let c2pa_table: Option<TableC2PA> = match sfnt.tables.get(&C2PA_TABLE_TAG) {
         None => None,
-        Some(ostensible_c2pa_table) => match ostensible_c2pa_table {
-            Table::C2PA(bonafied_c2pa_table) => Some(bonafied_c2pa_table.clone()),
-            _ => {
-                // Non-C2PA table with C2PA tag
-                return Err(Error::FontLoadError);
-            }
-        },
+        // If there is, replace its `manifest_store` value with the
+        // provided one.
+        Some(Table::C2PA(c2pa)) => Some(c2pa.clone()),
```

Comment:
Looks like we clone this just to check if it's okay later - could we just do `ok()` here instead and not do `c2pa_table.ok_or` later?